test/elm/list/list-test.ts

```diff
@@ -671,37 +672,51 @@ describe('First', () => {
 
   it('should get first of a list of numbers and the localId should exist on the root context', async function () {
     (await this.numbers.exec(this.ctx)).should.equal(1);
-    should(this.ctx.localId_context[6]).not.be.undefined();
+    const listLocalId = getLocalIdByPath(data, 'First', 'Numbers', 'source');
+    should(listLocalId).not.be.null();
```

Comment:
This is not harmful, but is it necessary? I assume that if `listLocalId` is `null`, then won't the test still fail on the next line anyway?

That said, I suppose it wouldn't be immediately obvious that it failed because it couldn't find the localId. But you could just have `getLocalIdByPath` throw an error if it can't find the id -- then the test would fail in a more obvious way when that happens.

Anyway... just a thought to possibly avoid having all these additional assertions.

Reply:
Good call! Implemented in [f0affad](https://github.com/cqframework/cql-execution/pull/320/commits/f0affadfd60e7b2892e34047f88a6b0fd6107ae8) and cleaned up the `not.be.null()` checks.